### PR TITLE
Upload Packages to PyPI

### LINKED ISSUE #2: Upload packages to pypi
See https://github.com/frequenz-floss/frequenz-api-microgrid-proto/issues/1 for more details.

### PR DESCRIPTION
**Pull request by leandro-lucarella-frequenz**
Upload packages to PyPI and improve CI:

- ci: Run on all branches
- ci: Upgrade GitHub action versions
- ci: Remove unnecessary credentials from checkout
- ci: Improve formatting
- ci: Remove unnecessary `fail-fast` config
- Move project metadata/config to pyproject.toml
- ci: Take into account the python version for caching
- Add extra metadata to pyproject.toml
- Exclude supporting files from the distribution
- ci: Publish distribution files to PyPI
- ci: Create GitHub Release automatically
- Create contributing guide
- Fix issue templates

If when upgrading you see this error:

> ERROR: Project file:///home/luca/devel/frequenz-sdk-python has
> a 'pyproject.toml' and its build backend is missing the 'build_editable'
> hook. Since it does not have a 'setup.py' nor a 'setup.cfg', it cannot be
> installed in editable mode. Consider using a build backend that supports PEP
> 660.

You might need to update the dependencies in you virtual environment (for
example `python -m pip install -U -e .`)

Fixes #2.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -34,42 +34,51 @@ jobs:
...
+        uses: actions/checkout@v3
 
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      with:
-        python-version: 3.8
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
```
Comment (jakub-toptal) [inline .github/workflows/ci.yaml:52]: Why change `3.8` to `3.10`? Is it the python version for which this job will publish `sdk` to PyPi?

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -9,3 +9,17 @@ python -m pip install frequenz-sdk
 ```
 
 The GitHub issues were also improved, adding templates for [reporting issues](https://github.com/frequenz-floss/frequenz-sdk-python/issues/new?assignees=&labels=priority%3A%E2%9D%93%2C+type%3Abug&template=bug.yml) and [requesting features](https://github.com/frequenz-floss/frequenz-sdk-python/issues/new?assignees=&labels=part%3A%E2%9D%93%2C+priority%3A%E2%9D%93%2C+type%3Aenhancement&template=feature.yml). Users are also pointed to the [Discussion forums](https://github.com/frequenz-floss/frequenz-sdk-python/issues/new/choose) when trying to open an issue if they have questions instead. Also many labels are assigned automatically on issue and pull request creation.
+
+## Upgrading
+
+Even if there are no breaking changes, if when upgrading you see this error in
+your local environment:
```
Comment (jakub-toptal) [inline RELEASE_NOTES.md]: > Even if there are no breaking changes, if when upgrading you see this error in
your local environment.

Even if there are no breaking changes, if you see this error in
your local environment while upgrading:

**Review by jakub-toptal — DISMISSED** [review]
LGTM (just some minor comments)

File: .github/RELEASE_NOTES.template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Release Notes
+
+## Summary
+
+<!-- Here goes a general summary of what this release is about -->
+
+## Upgrading
+
+<!-- Here goes notes on how to upgrade from previous versions, including if there are any depractions and what they should be replaced with -->
```
Comment (sahas-subramanian-frequenz) [inline .github/RELEASE_NOTES.template.md]: did you mean deprecation?

Comment (llucax) [reply .github/RELEASE_NOTES.template.md]: Yes. Will fix.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (llucax) [reply .github/workflows/ci.yaml:52]: No particular reason, it was mainly copy &paste from the channels repo. The python version is not "pinned" in the built packages so it doesn't really matter.

File: RELEASE_NOTES.md (hunk 2 above)
Comment (llucax) [reply RELEASE_NOTES.md]: Yup, it sounds weird. I will fix.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (leandro-lucarella-frequenz) [reply .github/workflows/ci.yaml:52]: I added a comment to the commit message.

File: RELEASE_NOTES.md (hunk 2 above)
Comment (leandro-lucarella-frequenz) [reply RELEASE_NOTES.md]: Done

File: .github/RELEASE_NOTES.template.md (hunk 3 above)
Comment (leandro-lucarella-frequenz) [reply .github/RELEASE_NOTES.template.md]: Done

**Comment by leandro-lucarella-frequenz** [comment]
Updated

**Review by sahas-subramanian-frequenz — APPROVED** [review]

**Pull request merged**